public/components/integrations/components/setup_integration.tsx

```diff
@@ -231,16 +296,64 @@ export function SetupBottomBar({
             isLoading={loading}
             onClick={async () => {
               setLoading(true);
-              const template = await findTemplate(integration.name);
-              await addIntegrationRequest(
-                false,
-                integration.name,
-                config.displayName,
-                template,
-                setToast,
-                config.displayName,
-                config.connectionDataSource
-              );
+              // Not sure why if I make an incrementer it doesn't change the prop,
+              // But since it doesn't, we track our progress manually.
+              let progress = loadingProgress;
+
+              if (config.connectionType === 'index') {
+                await addIntegrationRequest(
+                  false,
+                  integration.name,
+                  config.displayName,
+                  integration,
+                  setToast,
+                  config.displayName,
+                  config.connectionDataSource
+                );
+                progress += 2;
+                setProgress(progress);
+              } else if (config.connectionType === 's3') {
+                console.log('Starting S3 loading');
+                const http = coreRefs.http!;
+
+                console.log('Fetching assets');
+                const assets = await http.get(
+                  `${INTEGRATIONS_BASE}/repository/${integration.name}/assets`
+                );
+                progress += 1;
+                setProgress(progress);
+
+                console.log('Beginning queries');
+                // Queries must exist because we disable s3 if they're not present
+                for (const query of assets.data.queries!) {
+                  console.log('Query:', query);
+                  const queryStr = query.query.replace('${TABLE}', config.connectionDataSource);
+                  const result = await runQuery(queryStr, (step) => setProgress(progress + step));
+                  if (!result.ok) {
+                    console.error('Query failed', result.error);
+                    setLoading(false);
+                    setToast('Something went wrong.', 'danger');
```

Comment:
can we change this toast to more appropriate wording?

Reply:
+1

Reply:
What's more appropriate wording? We don't have any error information in the response that we can use